torch/distributed/pipelining/schedules.py

```diff
@@ -1140,8 +1175,14 @@ def _get_1f1b_rank_ops(
         n_local_stages * pp_group_size + 2 * (pp_group_size - 1 - rank)
     ) - (warmup_ops + rank)
 
+    if enable_zero_bubble:
```

Comment:
code cleanlines-wise i think zero bubble logic should probably just be in its own function rather than as part of `_get_1f1b_rank_ops`

Reply:
Hmm yeah makes sense. My following diff implements the other ZB algorithms and makes more changes here, lemme address this there if you are okay with it.